Source/EasyNetQ/Consumer/ConsumerFactory.cs

```diff
@@ -58,7 +58,7 @@ IConsumerConfiguration configuration
             IPersistentConnection connection,
             IConsumerConfiguration configuration)
         {
-            if (queue.IsExclusive)
+            if (queue.IsExclusive && configuration.RecoveryAction == null)
```

Comment:
if RecoveryAction is set to other than null, it means that we want to recover the subscription anyway. An example:
if we use nondurable exclusive queue for events, which is bind to a topic exchange, we definitly don't want to loose the subscription/queue in a cluster node failure, so we force it to be persistent consumer.